rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp

```diff
@@ -15,15 +15,103 @@
 #ifndef RMW_CONNEXT_SHARED_CPP__WAIT_HPP_
 #define RMW_CONNEXT_SHARED_CPP__WAIT_HPP_
 
+#include <unordered_map>
+#include <unordered_set>
+
 #include "ndds_include.hpp"
 
 #include "rmw/error_handling.h"
 #include "rmw/impl/cpp/macros.hpp"
 #include "rmw/types.h"
 
 #include "rmw_connext_shared_cpp/condition_error.hpp"
+#include "rmw_connext_shared_cpp/event_converter.hpp"
 #include "rmw_connext_shared_cpp/types.hpp"
 #include "rmw_connext_shared_cpp/visibility_control.h"
+#include "rmw_connext_shared_cpp/connext_static_event_info.hpp"
+
+
+rmw_ret_t __gather_event_conditions(
```

Comment:
nitpick: Conventionally we wrap the return type first if a declaration is too long, and only after that do we wrap parameters.

Specifically, I'm suggesting:

```c++
rmw_ret_t
__gather_event_conditions(
  rmw_events_t * events,
  std::unordered_set<DDS::StatusCondition *> & status_conditions)
{
  ...
}
```

Personally, I find it easier to read as well, because then the function names are always in the same column through the file.